pallets/ocex/src/tests.rs

```diff
@@ -20,113 +20,605 @@ use frame_support::{
+				AssetId::polkadex,
+				100_u128.into()
+			)
+		);
+	});
+}
+
+#[test]
+fn test_open_trading_pair_both_assets_cannot_be_same(){
+	new_test_ext().execute_with(||{
+		assert_noop!(
+			OCEX::open_trading_pair(
+				Origin::root(),
+				AssetId::asset(10),
+				AssetId::asset(10)
+			),
+			Error::<Test>::BothAssetsCannotBeSame
+		);
+	});
+}
+
+#[test]
+fn test_open_trading_pair_trading_pair_not_found(){
+	new_test_ext().execute_with(|| {
+		assert_noop!(
+			OCEX::open_trading_pair(
+				Origin::root(), 
+				AssetId::asset(10), 
+				AssetId::asset(20)
+			),
+			Error::<Test>::TradingPairNotFound
+		);
+	});
+}
+
+#[test]
+fn test_open_trading_pair(){
+	new_test_ext().execute_with(||{
+		assert_ok!(
+			OCEX::register_trading_pair(
+				Origin::root(), 
+				AssetId::asset(10), 
+				AssetId::asset(20), 
+				1_u128.into(),
+				100_u128.into(), 
+				1_u128.into(), 
+				100_u128.into(),
+				100_u128.into(),
+				10_u128.into()
+			)
+		);
+		assert_ok!(
+			OCEX::open_trading_pair(
+				Origin::root(),
+				AssetId::asset(10),
+				AssetId::asset(20)
+			)
+		);
+		assert_eq!(
+			TradingPairsStatus::<Test>::get(AssetId::asset(10), AssetId::asset(20)), 
+			true
+		);
+	})
+}
+
+#[test]
+fn test_close_trading_pair_both_assets_cannot_be_same(){
+	new_test_ext().execute_with(||{
+		assert_noop!(
+			OCEX::close_trading_pair(
+				Origin::root(),
+				AssetId::asset(10),
+				AssetId::asset(10)
+			),
+			Error::<Test>::BothAssetsCannotBeSame
+		);
+	});
+}
+
+#[test]
+fn test_close_trading_trading_pair_not_found(){
+	new_test_ext().execute_with(||{
+		assert_noop!(
+			OCEX::close_trading_pair(
+				Origin::root(),
+				AssetId::asset(10),
+				AssetId::asset(20)
+			),
+			Error::<Test>::TradingPairNotFound
+		);
+	});
+}
+
+#[test]
+fn test_close_trading_pair(){
+	new_test_ext().execute_with(||{
+		assert_ok!(
+			OCEX::register_trading_pair(
+				Origin::root(), 
+				AssetId::asset(10), 
+				AssetId::asset(20), 
+				1_u128.into(),
+				100_u128.into(), 
+				1_u128.into(), 
+				100_u128.into(),
+				100_u128.into(),
+				10_u128.into()
+			)
+		);
+		assert_ok!(
+			OCEX::close_trading_pair(
+				Origin::root(),
+				AssetId::asset(10),
+				AssetId::asset(20)
+			)
+		);
+		assert_eq!(
+			TradingPairsStatus::<Test>::get(AssetId::asset(10), AssetId::asset(20)), 
+			false
+		);
+	})
+}
+
+#[test]
+fn collect_fees(){
+	let account_id = create_account_id();
+	new_test_ext().execute_with(||{
+		// TODO! Discuss if this is expected behaviour, if not then could this be a potential DDOS?
+		assert_ok!(
+			OCEX::collect_fees(
+				Origin::signed(account_id.clone().into()),
+				100,
+				account_id.into()
+			)
+		);
+	});	
+}
+
+// P.S. This was to apply a DDOS attack and see the response in the mock environment
+/* #[test]
+fn collect_fees_ddos(){
+	let account_id = create_account_id();
+	new_test_ext().execute_with(||{
+		// TODO! Discuss if this is expected behaviour, if not then could this be a potential DDOS?
```

Comment:
Collect fees are callable only by the operator team and also it has a txn fee so, I don't think it can be DDoSed without incurring huge fees. What do you think? @felixfaisal 

cc: @zktony @35359595

Reply:
It needs to to be updated, I think @zktony has already created a PR.